lxd/patches.go

```diff
@@ -293,26 +295,30 @@ func patchStorageApi(name string, d *Daemon) error {
 
 			err = dbProfileConfigClear(tx, pID)
 			if err != nil {
+				shared.LogWarnf("Could not clear old profile configuration.")
 				tx.Rollback()
-				return err
+				continue
 			}
 
 			err = dbProfileConfigAdd(tx, pID, p.Config)
 			if err != nil {
+				shared.LogWarnf("Could not add new profile configuration.")
 				tx.Rollback()
-				return err
+				continue
 			}
 
 			err = dbDevicesAdd(tx, "profile", pID, p.Devices)
 			if err != nil {
+				shared.LogWarnf("Could not add new profile devices.")
 				tx.Rollback()
-				return err
+				continue
 			}
 
 			err = tx.Commit()
 			if err != nil {
+				shared.LogWarnf("Could not commit database transaction.")
 				tx.Rollback()
-				return err
+				continue
```

Comment:
Those errors should be more user friendly and should be LogErrorf as they are actual failure.

    "Failed to add a root device to profile \"%s\": %v"

Should do the trick so we can get users to grep their logs and tell us what profile failed.